improver/precipitation_type/shower_condition_probability.py

```diff
@@ -0,0 +1,182 @@
+            (convection,) = cubes.extract(self.convection_constraint)
+        except ValueError:
+            input_cubes = ", ".join([cube.name() for cube in cubes])
+            msg = (
+                "A cloud area fraction and convective ratio are required, "
+                f"but the inputs were: {input_cubes}"
+            )
+            raise ValueError(msg)
+
+        if cloud.shape != convection.shape:
+            msg = (
+                "The cloud are fraction and convective ratio cubes are not "
+                "the same shape and cannot be combined to generate a shower"
+                " probability"
+            )
+            raise ValueError(msg)
+        return cloud, convection
+
+    def process(self, cubes: CubeList) -> Cube:
+        """
+        Create a shower condition probability from cloud fraction and convective
+        ratio fields. This plugin thresholds the two input diagnostics,
+        creates a hybrid probability field from the resulting binary fields,
+        and then collapses the realizations to give a non-binary probability
+        field that represents the likelihood of conditions being showery.
+
+        Args:
+            cubes:
+                A cubelist containing a cube of cloud fraction and one of
+                convective ratio.
+
+        Returns:
+            Probability of any precipitation being classified as showery
+
+        Raises:
+            ValueError: If inputs are not those expected.
+            ValueError: If the input cubes have different shapes, perhaps due
+                        to a missing realization in one and not the other.
+        """
+        cloud, convection = self._extract_inputs(cubes)
+
+        # Threshold cubes
+        cloud_thresholded = BasicThreshold(
+            self.cloud_threshold, comparison_operator="<="
+        ).process(cloud)
+        convection_thresholded = BasicThreshold(self.convection_threshold).process(
+            convection
+        )
+
+        # Fill any missing data in the convective ratio field with zeroes.
+        if np.ma.is_masked(convection_thresholded.data):
+            convection_thresholded.data = convection_thresholded.data.filled(0)
+
+        # Create a combined field taking the maximum of each input
+        shower_probability = np.maximum(
+            cloud_thresholded.data, convection_thresholded.data
+        ).astype(FLOAT_DTYPE)
+
+        # Create a new diagnostic cube containing the new data
+        template, attributes = self._output_metadata(convection_thresholded)
+        result = create_new_diagnostic_cube(
+            template.name(),
+            "1",
+            template,
+            mandatory_attributes=attributes,
+            data=shower_probability,
+        )
+
+        try:
+            shower_conditions = iris.util.squeeze(collapse_realizations(result))
+        except CoordinateNotFoundError:
+            shower_conditions = iris.util.squeeze(result)
+
+        return shower_conditions
```

Comment:
```suggestion
        if "realization" in [c.name() for c in result.dim_coords()]:
            result = collapse_realizations(result)

        return iris.util.squeeze(result)

```

Reply:
But python.... https://devblogs.microsoft.com/python/idiomatic-python-eafp-versus-lbyl/

Reply:
Your link gives a specific example where your code is being too broad. I know that `iris.util.squeeze` isn't going to throw a `CoordinateNotFoundError`, but it doesn't belong in the `try` block as it is only the `collapse_realizations` function that you are checking. It should be in a `finally` block, or possibly after the whole construct.

Reply:
Agreed, and I so seldom use the `finally` option, meaning I am secretly (until this rash admission) delighted.

Reply:
Oh, but it really doesn't need to be in a finally clause. I got so excited about using it.... I'll just stick it at the end.